### PR TITLE
PYTHON-4243 Use GitHub App to Auto Assign Reviewer

### PR DESCRIPTION
**Pull request by blink1073**
Depends on https://github.com/mongodb-labs/drivers-evergreen-tools/pull/412

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -98,7 +98,7 @@ functions:
             # If this was a patch build, doing a fresh clone would not actually test the patch
             cp -R ${PROJECT_DIRECTORY}/ $DRIVERS_TOOLS
           else
-            git clone https://github.com/mongodb-labs/drivers-evergreen-tools.git $DRIVERS_TOOLS
+            git clone --branch add-reviewers https://github.com/blink1073/drivers-evergreen-tools.git $DRIVERS_TOOLS
```
Comment (Jibola) [inline .evergreen/config.yml:101]: could we have an `add-reviewers` branch directly tied to drivers-evergreen-tools?

File: .github/reviewers.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# List of reviewers for auto-assignment of reviews.
+caseyclements
+blink1073
+Jibola
+NoahStapp
```
Comment (Jibola) [inline .github/reviewers.txt:5]: Should shane be here as well?

**Review by Jibola — CHANGES_REQUESTED** [review]
Questions. Other than that, LGTM. :)

File: .evergreen/config.yml (hunk 1 above)
Comment (blink1073) [reply .evergreen/config.yml:101]: I'm going to add docs and open a PR once https://github.com/mongodb-labs/drivers-evergreen-tools/pull/411 is merged.

File: .github/reviewers.txt (hunk 2 above)
Comment (blink1073) [reply .github/reviewers.txt:5]: No, he's already exempt from the existing auto-assign.

**Comment by Jibola** [comment]
https://spruce.mongodb.com/task/mongo_python_driver_tests_python_version_rhel8_test_ssl__platform~rhel8_auth_ssl~auth_ssl_python_version~3.10_coverage~coverage_test_latest_sharded_cluster_patch_dc50cbd35ad74200d1f8eb9bbba1e3a624b53514_65df1ccec9ec44736b8900c0_24_02_28_11_45_19?execution=0&sortBy=STATUS&sortDir=ASC

Is this failure related?

File: .evergreen/config.yml [hunk 3]
```diff
@@ -2051,6 +2051,22 @@ tasks:
...
+          params:
+            shell: "bash"
+            working_dir: src
+            script: |
+              ${PREPARE_SHELL}
+              set -x
+              export CONFIG=$PROJECT_DIRECTORY/.github/reviewers.txt
+              export SCRIPT="$DRIVERS_TOOLS/.evergreen/github_app/assign-reviewer.sh"
+              bash $SCRIPT -p $CONFIG -h ${github_commit} -o "mongodb" -n "mongo-python-driver"
+              echo '{"results": [{ "status": "PASS", "test_file": "Build", "log_raw": "Test completed"  } ]}' > ${PROJECT_DIRECTORY}/test-results.json
```
Comment (NoahStapp) [inline .evergreen/config.yml:2068]: What's the purpose of this line?

Comment (blink1073) [reply .evergreen/config.yml:2068]: This is what generates a "Pass" on the test tab.

**Comment by blink1073** [comment]
> Is this failure related?

It is not.

**Review by Jibola — APPROVED** [review]
LGTM

**Pull request merged**